### DISCUSSION TITLE
setValue with dynamic path in typescript

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#5095] [Q&A] [answered]
Hi all, does anyone know how I can use `setValue` with a path that includes a variable? I want to set a field array value but I want to use a variable index. So for example:

```
setValue(`names.${index}`, { name: "mike" });
```

Unfortunately, this gives me a TS error. Does anyone know how I could do something like this?

https://codesandbox.io/s/distracted-banzai-tw8sg?file=/src/App.tsx

**Comment by leapful** [comment, marked as answer]
@wdfinch 

You need to cast the name as "const", it's a limitation with newest Typescript version.

Working sample: https://codesandbox.io/s/youthful-colden-svw7e

**Reply by wdfinch** [reply]
Thanks!

**Reply by wdfinch** [reply]
@leapful how could you get this to work if you're using typescript enums as your form keys? When I try the same operation using an enum to represent my keys, I get an error.

`Type 'string' is not assignable to type 'never'`

https://codesandbox.io/s/unruffled-wing-fufb0?file=/src/App.tsx

**Reply by wdfinch** [reply]
I'm gonna open this an issue since I believe this is related to the library.
